client/runtime/src/test/java/io/quarkiverse/openapi/generator/providers/UrlPatternMatcherTest.java

```diff
@@ -32,7 +39,16 @@ private static Stream<Arguments> providePathsThatMatch() {
                 Arguments.of("/{id}/{foo}/{id2}", "/1/2/3?q=1&q2=2"),
                 Arguments.of("/{id}/{foo}/{id2}", "/1/2/3"),
                 Arguments.of("/v2/pets/{id}", "/v2/pets/1"),
-                Arguments.of("/pets/{pet-id}/types/{type-id}", "/pets/1/types/2"));
+                Arguments.of("/pets/{pet-id}/types/{type-id}", "/pets/1/types/2"),
+                Arguments.of("/repos/{ref}", "/repos/prefixed/cool.sw"),
+                Arguments.of("/repos/{ref}/pepe", "/repos/prefixed/cool.sw/pepe"),
+                Arguments.of("pepe/pepa/pepu", "pepe/pepa/pepu"));
+    }
+
+    private static Stream<Arguments> providePathsThatNotMatch() {
+        return Stream.of(
+                Arguments.of("/pets/{id}", "/pes/1"),
+                Arguments.of("/{id}/pepe", "/1/2/pep"));
```

Comment:
In this case it doesn't match because "pep" vs "pepe", but however
`Arguments.of("/{id}/pepe", "/1/2/pepe"));`
matches, is it correct? The placeholder should only contain one id, not two, wdyt?